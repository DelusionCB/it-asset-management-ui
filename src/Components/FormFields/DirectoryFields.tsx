import React, {useEffect, useState} from 'react';
import {Row} from 'reactstrap';
import {
    directoryDefaultValues,
    extendedDirectoryDataProps,
} from '../../Types/types.directories';
import TextField from '../CustomComponents/TextField';
import CSelect from '../CustomComponents/CustomSelect';
import {handleDirectoryChange, handleDirectorySelect, debouncedCheck} from '../../Utils/helpers';
import {useTranslation} from 'react-i18next';
import ActionButtons from '../ActionButtons/ActionButtons';

interface FieldsStates {
    isDisabled: (is: boolean) => void
    type: string
}

function DirectoryFields ({isDisabled, type}: FieldsStates): JSX.Element {
    const [values, setValues] = useState<extendedDirectoryDataProps>(directoryDefaultValues)
    const {t} = useTranslation()

    function handleChange (e: React.ChangeEvent<HTMLInputElement>): void {
        handleDirectoryChange(e, values, setValues);
    }

    function handleSelect (value: {} | [] | undefined | null, id: string): void {
        handleDirectorySelect(value, id, values, setValues);
    }

    useEffect(() => {
        // Invoke debouncedCheck with updated values and directoryDefaultValues
        debouncedCheck(values, directoryDefaultValues, isDisabled);

        // Cleanup function to cancel the debounce on unmount
        return () => {
            debouncedCheck.cancel();
        };
    }, [values]);

    return (
        <>
            <Row>
                <h1>{t('directory.create')}</h1>
            </Row>
            <Row>
                <h2>{t('directory.description')}</h2>
            </Row>
            <Row>
                <TextField
                    id='name'
                    disabled={false}
                    invalid={false}
                    label={'values.name'}
                    onChange={(e) => { handleChange(e); }}
                    placeholder={'placeholder.name'}
                    value={values.name}
                />
            </Row>
            <Row>
                <TextField
                    id={'description'}
                    disabled={false}
                    invalid={false}
                    type='textarea'
                    label={'values.description'}
                    onChange={(e) => { handleChange(e) }}
                    placeholder={'placeholder.description'}
                    value={values.description}
                />
            </Row>
            <Row>
                <h2>{t('directory.dependency')}</h2>
            </Row>
            <Row>
                <CSelect
                    label={'values.applications'}
                    endpoint='application'
                    id={'applications'}
                    placeholder={'placeholder.applications'}
                    disabled={false}
                    invalid={false}
                    isMulti={true}
                    value={values.applications}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <CSelect
                    label={'values.servers'}
                    endpoint='server'
                    id={'servers'}
                    placeholder={'placeholder.servers'}
                    disabled={false}
                    invalid={false}
                    isMulti={true}
                    value={values.servers}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <Row>
                <CSelect
                    label={'values.services'}
                    endpoint='service'
                    id={'services'}
                    placeholder={'placeholder.services'}
                    disabled={false}
                    invalid={false}
                    isMulti={true}
                    value={values.services}
                    onChange={(value, id) => { handleSelect(value, id); }}
                />
            </Row>
            <ActionButtons
                values={values}
                actions={['create']}
                type={type}
            />
        </>
    );
}

export default DirectoryFields;
