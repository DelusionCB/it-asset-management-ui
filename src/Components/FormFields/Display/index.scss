
.directory-wrapper {
    background-color: white;
    box-shadow: 2px 2px 2px rgba(255, 255, 255, 0.3),
    -2px -2px 2px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
}
