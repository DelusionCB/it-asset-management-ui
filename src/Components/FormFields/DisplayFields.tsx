import React from 'react';
import {Container, Row} from 'reactstrap';
import {
    extendedApplicationDataProps, extendedDirectoryDataProps,
    LicenseDataProps,
} from '../../Types/types.directories';

interface appProps {
    type: 'application'
    item: extendedApplicationDataProps
}

interface dirProps {
    type: 'directory'
    item: extendedDirectoryDataProps
}

interface lcnProps {
    type: 'license'
    item: LicenseDataProps
}

type combinedProps = appProps | dirProps | lcnProps

function DisplayFields ({type, item}: combinedProps): JSX.Element {
    console.log(item, type, 'item')

    function getFields (): JSX.Element {
        switch (type) {
            case 'application':
                return (
                    <Container className='directoryItem-page'>
                        <Row>
                            <h1>Sovellus</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Sovelluksen kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Luokittelu:</h3>
                            <p>{item.classification}</p>
                        </Row>
                        <Row>
                            <h3>Asiakkuus:</h3>
                            <p>{item.place_of_use}</p>
                        </Row>
                        <Row>
                            <h3>Sisältääkö järjestelmä henkilörekisterin:</h3>
                            <p>{item.person_register ? 'Kyllä' : 'Ei'}</p>
                        </Row>
                        <Row>
                            <h3>Henkilötietojen katselun lokitus:</h3>
                            <p>{item.personal_info_logging ? 'Kyllä' : 'Ei'}</p>
                        </Row>
                        <Row>
                            <h3>Asennuksen lisätiedot:</h3>
                            <p>{item.install_info}</p>
                        </Row>
                        <Row>
                            <h3>Avainsanat:</h3>
                            <p>{item.keywords}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Sovelluksen sidokset</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Asennettu palvelimelle:</h3>
                            {item.installed_server
                                ? <>
                                    <p>{item.installed_server.name}</p>
                                </>
                                : <>
                                    <p>Ei asennettu</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Liittyy palveluihin:</h3>
                            {item.service_dependency
                                ? item.service_dependency.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei palvelukytköksiä</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Integraatiot:</h3>
                            {item.integration
                                ? item.integration.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei integraatioita</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Lisensointi:</h3>
                            {item.license
                                ? <>
                                    <p>{item.license.name}</p>
                                    <p>{item.license.valid_from_date}</p> - <p>{item.license.valid_until_date}</p>
                                </>
                                : <>
                                    <p>Ei lisenssiä</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Riippuvaisuudet sovelluksiin / sovelluksista:</h3>
                            {item.application_dependency
                                ? <>
                                    <p>{item.application_dependency.name}</p>
                                </>
                                : <>
                                    <p>Ei riippuvaisuuksia</p>
                                </>
                            }
                        </Row>
                        <Row>
                            <hr />
                            <h2>Jatkuvuus, kriittisyys & palautumistiedot</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Päivitykset:</h3>
                            <p>{item.update_practice}</p>
                        </Row>
                        <Row>
                            <h3>Varmistuskäytännöt & valvonta:</h3>
                            <p>{item.security_practice_monitoring}</p>
                        </Row>
                        <Row>
                            <h3>Palautuskäytännöt & toipuminen:</h3>
                            <p>{item.recovery_practices_convalescence}</p>
                        </Row>
                        <Row>
                            <h3>Käyttöoikeuksien hallinta:</h3>
                            <p>{item.user_rights_management}</p>
                        </Row>
                        <Row>
                            <h3>Tietoturvaratkaisut:</h3>
                            <p>{item.security_solutions}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Vastuut</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Sovelluksen omistaja:</h3>
                            <p>{item.product_owner}</p>
                        </Row>
                        <Row>
                            <h3>Sovelluksen haltija tai muu ensisijainen vastuuhenkilö:</h3>
                            <p>{item.application_holder}</p>
                        </Row>
                        <Row>
                            <h3>Pääkäyttäjät (Tekniset pääkäyttäjät):</h3>
                            <p>{item.admin_users}</p>
                        </Row>
                        <Row>
                            <h3>Vastuu & ammattikäyttäjät (Toiminnalliset pääkäyttäjät, tai esim. Pegasos vastuukäyttäjät):</h3>
                            <p>{item.liability_professional_users}</p>
                        </Row>
                        <Row>
                            <h3>Lisätietoja vastuuhenkilöistä:</h3>
                            <p>{item.holder_extra_info}</p>
                        </Row>
                        <Row>
                            <h3>Käyttöoikeusvastaavat:</h3>
                            <p></p>
                        </Row>
                        <Row>
                            <h3>Toimittaja:</h3>
                            <p>{item.provider}</p>
                        </Row>
                        <Row>
                            <h3>Toimittajan vastuut:</h3>
                            <p>{item.provider_responsibility}</p>
                        </Row>
                        <Row>
                            <h3>Lisäkontaktit:</h3>
                            <p>{item.additional_contacts}</p>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Lisätiedot & linkit</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Lisätietoja:</h3>
                            <p>{item.known_issues}</p>
                        </Row>
                        <Row>
                            <h3>Linkit & dokumentit:</h3>
                            <a>{item.fileUrl}</a>
                        </Row>
                    </Container>
                )
            case 'directory':
                return (
                    <Container>
                        <Row>
                            <h1>Kansio</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Kansion kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Sovellukset:</h3>
                            {item.applications.length
                                ? item.applications.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Palvelut:</h3>
                            {item.services.length
                                ? item.services.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                        <Row>
                            <h3>Palvelimet:</h3>
                            {item.servers.length
                                ? item.servers.map((child, key) => (
                                    <div key={key}>
                                        <Row>
                                            <h4>Nimi {key + 1}:</h4>
                                            <p>{child.name}</p>
                                        </Row>
                                        <Row>
                                            <h4>Kuvaus {key + 1}:</h4>
                                            <p>{child.description}</p>
                                        </Row>
                                    </div>
                                ))
                                : <>
                                    <h4>Ei sovelluksia</h4>
                                </>
                            }
                        </Row>
                    </Container>
                )
            case 'license':
                return (
                    <Container>
                        <Row>
                            <h1>Lisenssi</h1>
                        </Row>
                        <Row>
                            <hr />
                            <h2>Lisenssin kuvaus</h2>
                            <hr />
                        </Row>
                        <Row>
                            <h3>Nimi:</h3>
                            <p>{item.name}</p>
                        </Row>
                        <Row>
                            <h3>Kuvaus:</h3>
                            <p>{item.description}</p>
                        </Row>
                        <Row>
                            <h3>Voimassa alkaen:</h3>
                            <p>{item.valid_from_date}</p>
                        </Row>
                        <Row>
                            <h3>Voimassa päättyen:</h3>
                            <p>{item.valid_until_date}</p>
                        </Row>
                        <Row>
                            <h3>Lisenssityyppi:</h3>
                            <p>{item.license_type}</p>
                        </Row>
                        <Row>
                            <h3>Sopimus:</h3>
                            <p>{item.contract}</p>
                        </Row>
                        <Row>
                            <h3>Dokumentit:</h3>
                            <p>{item.fileUrl}</p>
                        </Row>
                    </Container>
                )
        }
    }
    return (
        <div>
            {getFields()}
        </div>
    );
}

export default DisplayFields;
