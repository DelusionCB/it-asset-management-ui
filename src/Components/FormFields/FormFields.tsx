import React, {useState} from 'react';
import ApplicationForm from './Forms/ApplicationForm';
import FormSelection from '../CustomComponents/FormSelection';
import {Container, Row} from 'reactstrap';
import DirectoryForm from './Forms/DirectoryForm';

function FormFields (): JSX.Element {
    const [fieldType, setFieldType] = useState('application')

    const [disabled, isDisabled] = useState(false)

    function getFormFields (): JSX.Element {
        switch (fieldType) {
            case 'application':
                return (
                    <ApplicationForm
                        type={fieldType}
                        isDisabled={isDisabled}
                    />
                )
            case 'directory':
                return (
                    <DirectoryForm
                        type={fieldType}
                        isDisabled={isDisabled}
                    />
                )
            case 'license':
                return (
                    <div />
                    // <LicenseForm
                    // type={fieldType}
                    // isDisabled={isDisabled} />
                )
            case 'server':
                return (
                    <div />
                    // <ServerForm type={fieldType} isDisabled={isDisabled} />
                )
            case 'service':
                return (
                    <div />
                    // <ServiceForm type={fieldType} isDisabled={isDisabled} />
                )
            default:
                return (<div />)
        }
    }

    return (
        <Container>
            <Row>
                <FormSelection
                    onChange={(e) => { setFieldType(e); }}
                    selections={['application', 'directory', 'license', 'server', 'service']}
                    disabled={disabled}
                />
            </Row>
            <Row>
                {getFormFields()}
            </Row>
        </Container>
    );
}

export default FormFields;
