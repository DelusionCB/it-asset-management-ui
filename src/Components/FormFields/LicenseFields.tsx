import React from 'react';
import {Row} from 'reactstrap';

function LicenseFields (): JSX.Element {
    return (
        <>
            <Row>
                <h1>Luo uusi lisenssi</h1>
            </Row>
            <Row>
                <h2>Lisenssin kuvaus</h2>
            </Row>
        </>
    );
}

export default LicenseFields;
