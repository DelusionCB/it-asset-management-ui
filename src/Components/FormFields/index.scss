.display-fields {
    h1, h2, h3, h4, h5 {
        color: var(--primary-color);
    }
    p {
        font-size: 1.25rem;
    }
}
