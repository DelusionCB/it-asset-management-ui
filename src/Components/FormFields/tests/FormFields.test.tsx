import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import FormFields from '../FormFields';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'

describe('<FormFields />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <FormFields mode='create' params={{}} loading={false} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
