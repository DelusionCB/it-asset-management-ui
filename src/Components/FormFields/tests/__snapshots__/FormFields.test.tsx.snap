// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFields /> Snapshot renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="selection-wrapper"
    >
      <div
        className="selection-header"
      >
        <h1>
          select-archive
        </h1>
      </div>
      <div
        className="selection-radios"
      >
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="application-0"
            >
              radios.application
            </label>
            <input
              aria-disabled={false}
              checked={true}
              className="form-check-input"
              disabled={false}
              id="application-0"
              name="application"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="directory-1"
            >
              radios.directory
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="directory-1"
              name="directory"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="license-2"
            >
              radios.license
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="license-2"
              name="license"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="server-3"
            >
              radios.server
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="server-3"
              name="server"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="service-4"
            >
              radios.service
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="service-4"
              name="service"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="integration-5"
            >
              radios.integration
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="integration-5"
              name="integration"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
        <div>
          <div
            className="form-check"
          >
            <label
              className="form-check-label"
              htmlFor="provider-6"
            >
              radios.provider
            </label>
            <input
              aria-disabled={false}
              checked={false}
              className="form-check-input"
              disabled={false}
              id="provider-6"
              name="provider"
              onChange={[Function]}
              type="radio"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  />
</div>
`;
